### ISSUE TITLE
Elasticsearch Connector - Nullable Struct fields

### ISSUE THREAD

**Issue by RossWilliams** [awslabs/aws-athena-query-federation#254]
I'm curious if it is intentional to force all defined fields in a structs to be defined in an elasticsearch document?

Returning null rather than throwing an exception would be preferred.

https://github.com/awslabs/aws-athena-query-federation/blob/b4ea654fba01a71ec86eacadefb1b22e35e62215/athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchFieldResolver.java#L74

**Comment by avirtuos** [comment]
@shurvitz I think this is a miss, we should be treating all these fields as nullable. at least until we support Apache Arrow's Map type. Thoughts?

**Comment by shurvitz** [comment]
@avirtuos @RossWilliams,
The Elasticsearch connector does not use schema inference. Rather, the table schema is well defined either by retrieving it directly from the Elasticsearch instance or from Glue (if used as a supplemental metadata source). So to answer your question, yes, the intent here is to ensure that the table schema matches the structure of the document for proper value extraction. **Notwithstanding, if your intent is to ignore specific columns in the schema, you can define a supplemental Glue table with a schema that contains only the columns you care about.**

In other words, this method only throws if your metadata definitions do not match the structure of the document returned from the Elasticsearch instance.

@RossWilliams, Do you have a use-case I can examine that demonstrates your point? Please include the JSON metadata definitions from the Elasticsearch instance (GET `<your table name>`/_mapping) and/or Glue table and sample data.

**Comment by avirtuos** [comment]
@shurvitz I'm not sure I agree with this. Let's connect offline and chat. I think @RossWilliams has a point about nullability. I can explain better with a quick call. Can you hit me up on slack when you are free?

**Comment by RossWilliams** [comment]
@shurvitz 
If I have a field in elasticsearch mapping:
```JSON
"attributes" : {
  "properties" : {
    "assessmentId" : {
      "type" : "text",
      "fields" : {
        "keyword" : {
          "type" : "keyword",
          "ignore_above" : 256
        }
      }
    }
  }
}
```
and a Glue Struct:
```
attributes: STRUCT <
  assessmentId: STRING
>
```

Not all of my documents in Elasticsearch will have an 'attributes.assessmentId' field. But the current connector code will throw an error if any documents in the connector result are missing 'attributes.assessmentId'

**Comment by avirtuos** [comment]
I connected with @shurvitz, he will be looking into making fields 'nullable' so that missing fields won't cause us to throw. This will hopefully be an easy change. stay tuned for updates.

**Comment by shurvitz** [comment]
PR #258 fixes this issue. Once merged you can build it locally and deploy: https://github.com/awslabs/aws-athena-query-federation/wiki/How_To_Build_A_Connector_or_UDF#upgrading
